Drivers/Src/Minions.c

```diff
@@ -0,0 +1,352 @@
+}
+
+/**
+ * @brief Toggles a light. Should be used only after Lights_Toggle_Set has been called for this light so that we are accurately tracking the enabled and disabled lights
+ * @param light Which light to toggle
+*/
+void Lights_Toggle(light_t light){
+    State lightState = Lights_Read(light);
+    if(lightState == OFF){
+        Lights_Set(light, ON);
+    } else {
+        Lights_Set(light, OFF);
+    }
+}
+
+/**
+
+ * @brief   Set light toggling
+ * @param   light Which light to enable toggling for
+ * @param   state State to set toggling
+ * @return  void
+ */
+void Lights_Toggle_Set(light_t light, State state) {
+    // Mutex not needed here because only BlinkLights uses this bitmap
+    if(light == LEFT_BLINK){
+        lightToggleBitmap |= 0x02;
+    }
+    else if(light == RIGHT_BLINK){
+        lightToggleBitmap |= 0x01;
+    }
+
+}
+
+/**
+* @brief   Read the state of a specific toggleable light from the toggle bitmap
+* @param   light Which Light to read
+* @return  returns State enum which indicates ON/OFF
+*/
+State Lights_Toggle_Read(light_t light) {
+    if(light == LEFT_BLINK){
+        return (State) (lightToggleBitmap&0x02 >> 1);
+    }
+    else if(light == RIGHT_BLINK){
+        return (State) (lightToggleBitmap&0x01);
+    }
+    else{
+        return OFF;
+    }
+}
+
+/**
+ * @brief   Read toggle bitmap
+ * @return  returns uint8_t bitmap for toggle (LEFT/RIGHT)
+ */
+uint8_t Lights_Toggle_Bitmap_Read(void) {
+    return lightToggleBitmap;
+
+}
+
+/**
+ * @brief   Set light to given state
+ * @param   light Which light to set
+ * @param   state What state to set the light to
+ * @return  void
+ */
+void Lights_Set(light_t light, State state) {
+    CPU_TS timestamp;
+    OS_ERR err;
+    State lightCurrentState = Lights_Read(light);
+    
+    if(lightCurrentState != state){     // Check if state has changed
+        uint8_t lightNewStates = lightStatesBitmap;
+        
+        lightNewStates &= ~(0x01 << light); // Clear bit corresponding to pin
+        lightNewStates |= (state << light); // Set value to inputted state
+        
+        // Initialize tx buffer and port c
+        uint8_t txWriteBuf[3] = {SPI_OPCODE_W, SPI_GPIOB, 0x00};
+        
+        if (light == BrakeLight) {  // Brakelight is only external
+            BSP_GPIO_Write_Pin(LIGHTS_PORT, BRAKELIGHT_PIN, ON);
+        } else {
+            txWriteBuf[2] = lightNewStates; // Write to tx buffer for lights present internally (on minion board)
+            
+            // Write to port c for lights present externally
+            switch (light) {
+                case LEFT_BLINK:
+                    BSP_GPIO_Write_Pin(LIGHTS_PORT, LEFT_BLINK_PIN, ON);
+                    break;
+                case RIGHT_BLINK:
+                    BSP_GPIO_Write_Pin(LIGHTS_PORT, RIGHT_BLINK_PIN, ON);
+                    break;
+                case Headlight_ON:
+                    BSP_GPIO_Write_Pin(LIGHTS_PORT, HEADLIGHT_PIN, ON);
```

Comment:
`switch (light) {
                case LEFT_BLINK:
                    BSP_GPIO_Write_Pin(LIGHTS_PORT, LEFT_BLINK_PIN, state);
                    break;
                case RIGHT_BLINK:
                    BSP_GPIO_Write_Pin(LIGHTS_PORT, RIGHT_BLINK_PIN, state);
                    break;
                case Headlight_ON:
                    BSP_GPIO_Write_Pin(LIGHTS_PORT, HEADLIGHT_PIN, state);`
Want to be able to turn it on and off, not only on